refactor: Remove unused boost signals2 from torcontrol

maflcko opened the pull request:
Remove unused boost, and other includes, and other legacy functions from torcontrol.

Commit:
Remove unused boost signals2 from torcontrol

Commit:
iwyu on torcontrol

Commit:
Replace LocaleIndependentAtoi with ToIntegral

No need for saturating behavior when the int is composed of 3 digits.

sedited reviewed (approved):
ACK fa32af22b323e7c58b6b20af6517f4795a72cdc5

The `async_handler` was unused since its introduction, so I don't think it's worth it to keep it around.

Commit:
remove unused limits.h include in compat.h

Commit:
Sort includes in compat.h

Can be reviewed with:
--color-moved=blocks  --color-moved-ws=ignore-all-space --ignore-all-space

dergoegge reviewed (approved):
utACK faaba770e11352ddf6414b9855f4baa46a967580

fanquake commented on src/torcontrol.h:
```diff
@@ -11,18 +11,14 @@
+#include <event2/util.h>
 
-#include <event2/bufferevent.h>
-#include <event2/event.h>
```
Is this (another) a bug in IWYU? Seems a bit weird to remove the `<event2/event.h>` include, which, as far as I can tell, is correct. It's where `event_base` is defined, and what is [expected when using libevent](https://github.com/libevent/libevent/blob/c9ec6aafb6a025f03636ae2ae7c18a2d4b8a7ed8/include/event2/event.h#L55):
>   section usage Standard usage
>  Every program that uses Libevent must include the <event2/event.h>
  header, and pass the -levent flag to the linker.

achow101 replied:
I think the since the libevent structs in this file are only pointers, the actual definitions aren't needed hence the include is not strictly necessary?

theuni replied:
Yeah, iirc everything in libevent can be forward-declared except for `evutil_socket_t`. It's very annoying.

Because everything else is already forward-declared as @achow101 mentioned, only the definition of `evutil_socket_t` is needed. `evutil_socket_t` is actually defined in `event2/util.h` (note that `event.h` includes `util.h`.

So I agree with this change. It's a small reduction in needed includes.

achow101 commented:
ACK faaba770e11352ddf6414b9855f4baa46a967580

atou69 reviewed (commented):
#27511
